### ISSUE TITLE
Why average loss value by `batch_size` when using `batch_hard` method?

### ISSUE THREAD

**Issue by Hongtao-Niro** [omoindrot/tensorflow-triplet-loss#55]
Hi Olivier,

Thanks for your work, it helped me a lot! Just one question: 

In the `batch_hard_triplet_loss` method, you calculate the average loss value using `tf.reduce_mean`, which is equivalent to `tf.reduce_sum(tensor) / len(tensor)`. As a result the losses are averaged over `batch_size`. Shouldn't the output loss value be the average over number of non-zero losses? For example, let's say `triplet_loss` at [line 218](https://github.com/omoindrot/tensorflow-triplet-loss/blob/fc698369bb6c9acdc9f0e9e1ea00de0ddf782f12/model/triplet_loss.py#L218) is `[1.4, 0, 0, 0 1.6]`, the output of the function will be 0.6 according to the code, but shouldn't it be 1.5?

I saw a [similar discussion](http://disq.us/p/1w7th2e) under your triplet loss post, but the code doesn't seem to reflect the thought.
